### PR TITLE
Add Command Line Open-As-Workspace Flag (Issue #4253)

### PR DESCRIPTION
**Pull request by TCWORLD**
In order to use Notepad++ as an editor for the new Github Desktop, support for opening a directory is required.

Notepad++ already has the ability to open a directory as a workspace, however this is not possible from the command line as the default behaviour is to open all files in any supplied directory.

In this commit, a new flag, "-dirAsWorkspace" has been added which will treat any directory in the CLI fileName input as a workspace and open it as such. Any files will be treated as files and opened as usual.

Basically this PR would allow https://github.com/desktop/desktop/pull/4120 to be completed and get Notepad++ back as an editor in Github Desktop.

### CODE REVIEW

File: PowerEditor/src/Notepad_plus.cpp [hunk 1]
```diff
@@ -5315,58 +5315,86 @@ void Notepad_plus::loadCommandlineParams(const TCHAR * commandLine, const CmdLin
...
+				}
+			}
 
-		if (lt != L_EXTERNAL && lt < nppParams->L_END)
-		{
-			Buffer * pBuf = MainFileManager->getBufferByID(bufID);
-			pBuf->setLangType(lt);
+			folderPaths.push_back(path);
 		}
+		else {
```
Comment (donho) [inline PowerEditor/src/Notepad_plus.cpp]: bad coding style regarding Notepad++ coding style:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

File: PowerEditor/src/Notepad_plus.cpp [hunk 2]
```diff
@@ -5315,58 +5315,86 @@ void Notepad_plus::loadCommandlineParams(const TCHAR * commandLine, const CmdLin
...
 	{
 		const TCHAR *pFn = fnss.getFileName(i);
 		if (!pFn) return;
 
-		BufferID bufID = doOpen(pFn, recursive, readOnly);
-		if (bufID == BUFFER_INVALID)	//cannot open file
-			continue;
-
-		lastOpened = bufID;
+		if (workspace && ::PathIsDirectory(pFn)) {
```
Comment (donho) [inline PowerEditor/src/Notepad_plus.cpp]: bad coding style regarding Notepad++ coding style:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

File: PowerEditor/src/Notepad_plus.cpp [hunk 3]
```diff
@@ -5315,58 +5315,86 @@ void Notepad_plus::loadCommandlineParams(const TCHAR * commandLine, const CmdLin
...
 	}
 
- 	LangType lt = pCmdParams->_langType;
-	int ln =  pCmdParams->_line2go;
-    int cn = pCmdParams->_column2go;
-    int cpos = pCmdParams->_pos2go;
+	LangType lt = pCmdParams->_langType;
+	int ln = pCmdParams->_line2go;
+	int cn = pCmdParams->_column2go;
+	int cpos = pCmdParams->_pos2go;
```
Comment (donho) [inline PowerEditor/src/Notepad_plus.cpp]: Make your modification compact - don't reformat source code in your request. It makes code review more difficult.
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests

**Comment by donho** [comment]
I really don't think you need to change the `loadCommandlineParams` fuction logic for adding this feaure.
